source/N4509.cc

```diff
@@ -0,0 +1,36 @@
+#include <atomic>
+#include <iostream>
+
+// It is implementation-defined what this returns, as long as:
+// if (atomic<T>::is_lock_free())
```

Comment:
Actually, the trait here needs a bit of expository wrapping to show how it's supposed to be used in std::atomic.  Maybe a prototype for std::atomic::is_always_lock_free in comments.